executor/aggregate.go

```diff
@@ -224,6 +225,13 @@ func (e *HashAggExec) Close() error {
 	for range e.finalOutputCh {
 	}
 	e.executed = false
+
+	// len(e.partialWorkers) == len(e.finalWorkers)
+	concurrency := len(e.partialWorkers)
```

Comment:
Why don't we set it in planner phase like previous version? The concurrency can be caculated in planner.

Reply:
Because of the ```projection  concurrency``` can not be calculated accurately in the planner phase. See [here](https://github.com/pingcap/tidb/blob/fea5ce1114c43d1ea4795c7eedfa209c4610b76a/executor/projection.go#L33-L39) and [here](https://github.com/pingcap/tidb/blob/fea5ce1114c43d1ea4795c7eedfa209c4610b76a/executor/projection.go#L87-L92) for more details.